### PR TITLE
Address review

### PR DESCRIPTION
**Pull request by FreeApophis**

### CODE REVIEW

File: Funcky.Money.SourceGenerator/Iso4217RecordGenerator.cs [hunk 1]
```diff
@@ -40,8 +40,8 @@ private static string GenerateCurrencyClass(IReadOnlyList<Iso4217Record> records
                $"{{{NewLine}" +
                $"{Indent}public partial record Currency{NewLine}" +
                $"{Indent}{{{NewLine}" +
-               $"{GenerateAllCurrenciesProperty(records)}" +
                $"{GenerateCurrencyProperties(records)}" +
+               $"{GenerateAllCurrenciesProperty(records)}" +
```
Comment (FreeApophis) [inline Funcky.Money.SourceGenerator/Iso4217RecordGenerator.cs:44]: All values in AllCurrencies were null due to the order of static initialisation

File: Funcky.Money.SourceGenerator/Iso4217RecordGenerator.cs [hunk 2]
```diff
@@ -86,7 +86,7 @@ private static string GenerateAllCurrenciesProperty(IReadOnlyCollection<Iso4217R
         {
             var property = new StringBuilder();
 
-            property.AppendLine($"{Indent}{Indent}internal static IReadOnlyCollection<Currency> AllCurrencies {{ get; }} = ImmutableArray.Create(");
+            property.AppendLine($"{Indent}{Indent}internal static ImmutableArray<Currency> AllCurrencies {{ get; }} = ImmutableArray.Create(");
```
Comment (FreeApophis) [inline Funcky.Money.SourceGenerator/Iso4217RecordGenerator.cs]: I want to access Length (which would in IReadOnlyCollection) and the index operator (which would be in IReadOnlyList) so we need to expose the type here.

Comment (tautropfli) [reply Funcky.Money.SourceGenerator/Iso4217RecordGenerator.cs]: There's `IReadOnlyList`…

File: Funcky.Money.SourceGenerator/Iso4217RecordGenerator.cs (hunk 1 above)
Comment (tautropfli) [reply Funcky.Money.SourceGenerator/Iso4217RecordGenerator.cs:44]: Silly me 😆

File: Funcky.Money/Exceptions/InvalidPrecisionException.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Runtime.Serialization;
+
+namespace Funcky
+{
+    [Serializable]
+    internal class InvalidPrecisionException : Exception
```
Comment (tautropfli) [inline Funcky.Money/Exceptions/InvalidPrecisionException.cs]: Could this inherit from `ArgumentException`?

**Review by tautropfli — CHANGES_REQUESTED** [review]

File: Funcky.Money.SourceGenerator/Iso4217RecordGenerator.cs (hunk 2 above)
Comment (FreeApophis) [reply Funcky.Money.SourceGenerator/Iso4217RecordGenerator.cs]: Adressed in 320b99b5

**Review by tautropfli — APPROVED** [review]

**Pull request merged**